ntp.go

```diff
@@ -364,13 +364,11 @@ func getTime(host string, opt QueryOptions) (*msg, ntpTime, error) {
 	// when the messsage was actually sent.
 	r := make([]byte, 8)
 	_, err = rand.Read(r)
-	var sendTime ntpTime
+	sendTime := time.Now()
 	if err == nil {
 		xmitMsg.TransmitTime = ntpTime(binary.BigEndian.Uint64(r))
-		sendTime = toNtpTime(time.Now())
 	} else {
-		xmitMsg.TransmitTime = toNtpTime(time.Now())
-		sendTime = xmitMsg.TransmitTime
+		xmitMsg.TransmitTime = toNtpTime(sendTime)
 	}
```

Comment:
I'd like to suggest a small order of operations change to your code:
```go
r := make([]byte, 8)
_, err = rand.Read(r)
var sendTime time.Time
if (err == nil) {
    xmitMsg.TransmitTime = ntpTime(binary.BigEndian.Uin64(r))
    sendTime = time.Now()
} else {
    sendTime = time.Now()
    xmitMsg.TransmitTime = toNtpTime(sendTime)
}
```

The reason I would like it done this way is so that there is as little code as possible between the call to `time.Now()` and the message to the NTP server.  That way the RTT measurement doesn't include the computation time of `ntpTime(binary.BigEndian.Uint64(r))`.